web/app/themes/justicejobs/inc/jobs-handler/add-edit-jobs.php

```diff
@@ -1,124 +1,189 @@
 <?php
+if (!function_exists('import_jobs_from_xml'))   {
 
-if (!function_exists('addJobPost'))   {
-
-function addJobPost() {
+function import_jobs_from_xml() {
   require_once( ABSPATH . 'wp-admin/includes/post.php' );
 
-  $allposts= get_posts( array('post_type'=>'job','numberposts'=>-1) );
-  foreach ($allposts as $eachpost) {
-    wp_delete_post( $eachpost->ID, true );
-  }
-
-
-
   $xml = simplexml_load_file( "app/uploads/job-feed/jobs.xml") or die("Fail");
 
   $totaljobs = count($xml->entry);
-  $job_id;
-  $job_title;
-  $job_link;
-  $salary;
-  $working_patterns = array();
-  $job_locations = array();
-
 
     for ($x = 0; $x < $totaljobs; $x++) {
-      $totalspans = count($xml->entry[$x]->content->div->span);
-      $job_id = $xml->entry[$x]->content->div->span[0];
+      $job_content =  $xml->entry[$x]->content;
+      $totalspans = count($job_content->div->span);
       $job_title = $xml->entry[$x]->title;
-      $job_link = (string) $xml->entry[$x]->id;
 
-      echo $job_id;
-      echo $job_title;
+      if($totalspans > 0) {
 
+          $job_id = (string) $job_content->div->span[0];
+          $job_link = (string) $xml->entry[$x]->id;
 
-      if (post_exists($job_title)) {
+          $job_content_hash = md5($job_content->div->asXML());
 
-        echo 'woot woot!';
+          $job_title = str_replace($job_id . " - ", "", $job_title);
 
-      } else {
+          $job_title = $job_title . " - " . $job_id;
+          echo $job_title;
+          echo "<br/>";
 
-        // Creates job post
-        $job_post = array(
-          'post_title'    => $job_title,
-          'post_content'  => ' ',
-          'post_status'   => 'publish',
-          'post_type'     => 'job'
-        );
+          $args = array(
+              'post_type'=>'job',
+              'numberposts'=>-1,
+              'meta_query' => array(
+              array(
+                  'key' => 'job_id',
+                  'value' => $job_id,
+                  'compare' => '=',
+              )
+          )
+          );
+          $job_check = get_posts( $args );
 
-        // Insert the post into the database
-        $post_id = wp_insert_post( $job_post );
+          if(count($job_check) == 0) {
 
-        for ($y = 0; $y < $totalspans; $y++) {
-          // Save Role Type Info
-          if ($xml->entry[$x]->content->div->span[$y]->attributes()->itemprop[0] == "Role Type") {
-            $role_type = (string) $xml->entry[$x]->content->div->span[$y];
+              /**
```

Comment:
Hey, can you remove the commented code? GH is complaining a little bit - cheers bud

Reply:
Done. Thanks Mate.